src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/build_task.py

```diff
@@ -19,148 +22,197 @@ def acr_build_task_create(cmd,
 
-def acr_build_task_run(
-    cmd,
-    client,
-    build_task_name,
-    registry_name,
-    no_logs=False,
-    resource_group_name=None):
+    if timeout is not None:
+        instance.timeout = timeout
+
+    if repository_url is not None:
+        instance.source_repository.repository_url = repository_url
+
+    if commit_trigger_enabled is not None:
+        instance.source_repository.is_commit_trigger_enabled = commit_trigger_enabled == 'true'
+
+    if git_access_token is not None:
+        instance.source_repository.source_control_auth_properties.git_access_token = git_access_token
+
+    if tags is not None:
+        instance.tags = tags
+
+    return instance
+
+
+def acr_build_task_run(cmd,
+                       client, # cf_acr_builds
+                       build_task_name,
+                       registry_name,
+                       no_logs=False,
+                       resource_group_name=None):
     _, resource_group_name = validate_managed_registry(
         cmd.cli_ctx, registry_name, resource_group_name, BUILD_TASKS_NOT_SUPPORTED)
 
     from ._client_factory import cf_acr_registries
     client_registries = cf_acr_registries(cmd.cli_ctx)
-    buildTaskBuildRequest = BuildTaskBuildRequest(build_task_name=build_task_name)
 
     queued_build = LongRunningOperation(cmd.cli_ctx)(
-        client_registries.queue_build(resource_group_name, registry_name, buildTaskBuildRequest))        
+        client_registries.queue_build(resource_group_name, registry_name, BuildTaskBuildRequest(build_task_name)))
+
     if no_logs:
         return queued_build
     else:
         if queued_build:
             build_id = queued_build.build_id
-            print("Queued a build with build-id: {}.".format(build_id))
-            print("Starting to stream the logs...")
+            print("Queued a build with build ID: {}.".format(build_id))
+            print("Waiting for a build agent")
             acr_build_task_logs(cmd, client, registry_name, build_id)
 
 
-def acr_build_task_logs(
-    cmd,
-    client,
-    registry_name,
-    build_id=None,
-    resource_group_name=None):
+def acr_build_task_list_builds(cmd,
+                               client, # cf_acr_builds
+                               registry_name,
+                               build_task_name=None,
+                               resource_group_name=None):
     _, resource_group_name = validate_managed_registry(
         cmd.cli_ctx, registry_name, resource_group_name, BUILD_TASKS_NOT_SUPPORTED)
 
-    if build_id is None:     
+    if build_task_name:
+        filter_str = "BuildTaskName eq '{}'".format(build_task_name)
+        return client.list(resource_group_name, registry_name, filter=filter_str)
+    else:
+        return client.list(resource_group_name, registry_name)
+
+
+def acr_build_task_logs(cmd,
+                        client, # cf_acr_builds
+                        registry_name,
+                        build_task_name=None,
```

Comment:
move this param after `build_id`, or change the call in line #180:
`acr_build_task_logs(cmd, client, registry_name, build_id)`

Reply:
Will rebase on https://github.com/AzureCR/azure-cli/pull/117